packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -1900,12 +1925,16 @@ export interface BuildEnvironmentVariable {
    * The value of the environment variable.
    * For plain-text variables (the default), this is the literal value of variable.
    * For SSM parameter variables, pass the name of the parameter here (`parameterName` property of `IParameter`).
-   * For SecretsManager variables secrets, pass either the secret name (`secretName` property of `ISecret`)
-   * or the secret ARN (`secretArn` property of `ISecret`) here,
-   * along with optional SecretsManager qualifiers separated by ':', like the JSON key, or the version or stage
-   * (see https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html#build-spec.env.secrets-manager for details).
+   * For SecretsManager variables secrets, pass either the secret name (`secretName` property of `ISecret`),
+   * the secret ARN (`secretArn` property of `ISecret`) or the secret itself (type of ISecret) here.
    */
   readonly value: any;
+
+  /**
+   * Optional SecretsManager qualifiers separated by ':',like the JSON key, or the version or stage,
+   * (see https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html#build-spec.env.secrets-manager for details).
+   */
+  readonly qualifier?: string
```

Comment:
Adding a qualifier field here to support json keys when passing a `Secret` seems like the best solution to me. Should we also change this when passing secret name or secret Arn to avoid confusion?

What do you think @skinny85 ?

Reply:
I'm actually not a huge fan of that... I think writing ``value: `${secret.secretArn}:json-key`,`` is much more natural than `value: secret.secretArn, qualifier: 'json-key',`...

Reply:
True. maybe adding a method to `SecretValue` which returns the secretArn could solve this.
Otherwise I don't see a different way of implementing json key support when passing a Secret other than the qualifier field.

If you don't like the qualifier field I can go and implement a secretArn field for the `SecretValue` and a getter for it. Then we would be able to use it instead of `Secret`.

What do you think?

Reply:
So, what type do you think this new `secretArn` property in `SecretValue` should have?

Reply:
I was thinking of a String to hold the value :)

Reply:
OK. Let's make it `readonly secretArn?: string` then.